core/trino-main/src/main/java/io/trino/operator/scalar/JsonOutputFunctions.java

```diff
@@ -0,0 +1,168 @@
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.operator.scalar.JsonConversionErrors.JsonOutputConversionError;
+import io.trino.spi.function.ScalarFunction;
+import io.trino.spi.function.SqlNullable;
+import io.trino.spi.function.SqlType;
+import io.trino.spi.type.StandardTypes;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.trino.sql.tree.JsonQuery.Behavior.EMPTY_ARRAY;
+import static io.trino.sql.tree.JsonQuery.Behavior.EMPTY_OBJECT;
+import static io.trino.sql.tree.JsonQuery.Behavior.ERROR;
+import static io.trino.sql.tree.JsonQuery.Behavior.NULL;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Format JSON as binary or character string, using given encoding.
+ * <p>
+ * These functions are used to format the output of JSON_QUERY function.
+ * In case of error during JSON formatting, the error handling
+ * strategy of the enclosing JSON_QUERY function is applied.
+ * <p>
+ * Additionally, the options KEEP / OMIT QUOTES [ON SCALAR STRING]
+ * are respected when formatting the output.
+ */
+public class JsonOutputFunctions
+{
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final EncodingSpecificConstants UTF_8 = new EncodingSpecificConstants(
+            JsonEncoding.UTF8,
+            StandardCharsets.UTF_8,
+            Slices.copiedBuffer(new ArrayNode(JsonNodeFactory.instance).asText(), StandardCharsets.UTF_8),
+            Slices.copiedBuffer(new ObjectNode(JsonNodeFactory.instance).asText(), StandardCharsets.UTF_8));
+    private static final EncodingSpecificConstants UTF_16 = new EncodingSpecificConstants(
+            JsonEncoding.UTF16_LE,
+            StandardCharsets.UTF_16LE,
+            Slices.copiedBuffer(new ArrayNode(JsonNodeFactory.instance).asText(), StandardCharsets.UTF_16LE),
+            Slices.copiedBuffer(new ObjectNode(JsonNodeFactory.instance).asText(), StandardCharsets.UTF_16LE));
+    private static final EncodingSpecificConstants UTF_32 = new EncodingSpecificConstants(
+            JsonEncoding.UTF32_LE,
+            Charset.forName("UTF-32LE"),
+            Slices.copiedBuffer(new ArrayNode(JsonNodeFactory.instance).asText(), Charset.forName("UTF-32LE")),
+            Slices.copiedBuffer(new ObjectNode(JsonNodeFactory.instance).asText(), Charset.forName("UTF-32LE")));
+
+    private JsonOutputFunctions() {}
+
+    @SqlNullable
+    @ScalarFunction(value = "$json_to_varchar", hidden = true)
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice jsonToVarchar(@SqlType(StandardTypes.JSON_2016) JsonNode jsonExpression, @SqlType(StandardTypes.TINYINT) long errorBehavior, @SqlType(StandardTypes.BOOLEAN) boolean omitQuotes)
+    {
+        return toSlice(jsonExpression, UTF_8, errorBehavior, omitQuotes);
+    }
+
+    @SqlNullable
+    @ScalarFunction(value = "$json_to_varbinary", hidden = true)
```

Comment:
drop and use `$json_to_varbinary_utf8` directly instead?

Reply:
I'd rather leave it, same as https://github.com/trinodb/trino/pull/9831#discussion_r841194339